### PR TITLE
[doc] add a doc page about ci testing workflow on a PR

### PR DESCRIPTION
**Pull request by can-anyscale**
As title, add a doc page about the new ci testing workflow (microcheck, go, premerge, etc.). I'll follow up with another PR to add more images to the doc.

Test:
- https://anyscale-ray--45446.com.readthedocs.build/en/45446/ray-contribute/ci.html

Screenshot 2024-05-20 at 11 02 29 AM

### CODE REVIEW

File: doc/source/ray-contribute/ci.rst
Comment (aslonnie) [inline doc/source/ray-contribute/ci.rst:44]: say will finish when all the tests pass?

do we want to explicitly say that only committers can merge?

File: doc/source/ray-contribute/ci.rst
Comment (aslonnie) [inline doc/source/ray-contribute/ci.rst:37]: I am not sure what the "summary" we mean here. do we mean the "status" of that commit?

File: doc/source/ray-contribute/ci.rst
Comment (aslonnie) [inline doc/source/ray-contribute/ci.rst:50]: should we want to mention that when the PR has new commit then the auto merge will be auto disabled?

File: doc/source/ray-contribute/ci.rst
Comment (aslonnie) [inline doc/source/ray-contribute/ci.rst:26]: maybe don't add `-a`? probably should not encourage / tell user to use `-a` here as if `-a` is required for the workflow.

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:4]: let me add the explanation for CI here

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:47]: i'm using the cost per postmerge run as an approximation, it stands between $50-60 before we move it to scheduled run

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:37]: status yes

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:50]: let's add that as well

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:26]: what is your favorite workflow to add a git commit message? the `-a` is required to add a new commit message (there are similar alternatives like `-m` but they will need something)

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:1]: good catch, i'm sticking to the most common styling then

**Review by aslonnie — APPROVED** [review]
please wait for doc reviews.

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [reply doc/source/ray-contribute/ci.rst:1]: Could you install Vale? Our style guide is to sentence case headings. Thanks! https://anyscale-ray--45446.com.readthedocs.build/en/45446/ray-contribute/docs.html

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:1]:
```suggestion
CI testing workflow on PRs
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:5]:
```suggestion
workflow on a PR. CI refers to the automated testing of the codebase
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:10]:
```suggestion
With every commit on your PR, by default, CI runs a set of tests
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:13]:
```suggestion
These tests are 90% accurate at catching bugs on your
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:15]:
```suggestion
microcheck typically runs twice as fast and costs half as much as
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:18]:
```suggestion
* If you add a new test or modify an existing test in a PR,
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:19]:
```suggestion
  microcheck includes these tests.
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:22]:
```suggestion
  `@microcheck TEST_TARGET01 TEST_TARGET02 ....`. You must put this line in the
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:23]:
```suggestion
  body of the message, on the second line or
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:24]:
```suggestion
  below. The first line is the commit message title. For example:
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:25]:
```suggestion
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:37]:
```suggestion
If microcheck passes, you see a green checkmark on your PR. If it
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:38]:
```suggestion
fails, you see a red cross. In either case, you see a summary of
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:39]:
```suggestion
the test run statuses in the GitHub UI.
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:44]:
```suggestion
To merge your PR in this workflow, click the **Enable auto-merge**
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:45]:
```suggestion
button, or ask a committer to do it. This action trigger sadditional test
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:46]:
```suggestion
cases, and the PR merges automatically once they finish and pass.
```

Comment (angelinalg) [reply doc/source/ray-contribute/ci.rst:46]: Does is trigger `additional tests` or `additional test cases`?

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:48]:
```suggestion
Alternatively, you can add the `go` label to manually trigger the full
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:49]:
```suggestion
test suite on your PR. Be mindful of the cost and run time consequences of this option.
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:50]:
```suggestion
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:51]:
```suggestion
You should only need to run the full suite in rare cases. However, if you require it frequently, 
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:52]:
```suggestion
let the Ray team know. The Ray team is continuously improving the effectiveness of microcheck.
```

File: doc/source/ray-contribute/ci.rst
Comment (angelinalg) [inline doc/source/ray-contribute/ci.rst:8]:
```suggestion
`microcheck`
```

**Review by angelinalg — APPROVED** [review]
Just some style nits. Please consider installing the Vale extension to catch these earlier. (go/vale) Thanks!

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:1]: will do thankks

File: doc/source/ray-contribute/ci.rst
Comment (can-anyscale) [reply doc/source/ray-contribute/ci.rst:46]: i updated to tests, thankks

**Pull request merged**
